rcl/include/rcl/type_hash.h

```diff
@@ -0,0 +1,76 @@
+// Copyright 2023 Open Source Robotics Foundation, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef RCL__TYPE_HASH_H_
+#define RCL__TYPE_HASH_H_
+
+#ifdef __cplusplus
+extern "C"
+{
+#endif
+
+#include "rcl/types.h"
+#include "rcl/visibility_control.h"
+#include "rcutils/sha256.h"
+#include "rosidl_runtime_c/type_hash.h"
+#include "type_description_interfaces/msg/type_description.h"
+
+/// Given a TypeDescription, output a string of the hashable JSON representation of that data.
+/**
+ * The output here is generally hashed via rcl_calculate_type_hash below.
+ * Compare this reference implementation with rosidl_parser.type_hash.idl_to_hashable_json.
+ * Both must produce the same output for the same types, providing a stable reference for
+ * external implementations of the ROS 2 Type Version Hash.
+ *
+ * The JSON representation contains all types and fields of the original message, but excludes:
+ * - Default values
+ * - Comments
+ * - The input plaintext files that generated the TypeDescription
+ *
+ * \param[in] type_description Prefilled TypeDescription message to be translated
+ * \param[out] output_repr An initialized empty char array that will be filled with
+ *   the JSON representation of type_description. Note that output_repr will have a
+ *   terminating null character, which should be omitted from hashing. To do so, use
+ *   (output_repr.buffer_length - 1) or strlen(output_repr.buffer) for the size of data to hash.
+ * \return RCL_RET_OK on success, RCL_RET_ERROR if any problems occur in translation
```

Comment:
nitpick: you might want to have a look at how the other rcl functions document return types, I think they use one per `\return` and wrap the return types in ``` `RCL_RET_...` ``` for a reason, but I can't remember them off-hand. Thought I do think it's best to just emulate them.

Reply:
Done (they use 
```
\return #RCL_RET_... for reason, or
\return #RCL_RET_... for a final reason, period.
``` 

so I did that